### PR TITLE
perf: optimize message delivery and give bs hints

### PR DESCRIPTION
**Pull request by isidentical**
Optimizes message delivery and handling (adds around ~5-6 FPS to the realtime turbo demo) by not blocking the event loop when processing the inputs and continously receiving/sending stuff in the background. Also changes dynamic batching API to give a hint of `current_batch_size` so the inputs themselves can also decide how long of a batch they permit.

### CODE REVIEW

File: projects/fal/src/fal/app.py [hunk 1]
```diff
@@ -232,13 +232,29 @@ async def mirror_input(queue: deque[Any], websocket: WebSocket) -> None:
...
+
         while True:
             if not queue:
                 await asyncio.sleep(0.05)
                 continue
 
             input = queue.popleft()
-            if input is None:
+            if input is None or emitter.done():
+                emitter.cancel()
```
Comment (squat) [inline projects/fal/src/fal/app.py:255]: This seems funny to me like the more idiomatic way would be to raise a `StopIteration` exception or something. Not a blocking concern for me though

Comment (isidentical) [reply projects/fal/src/fal/app.py:255]: raising a StopIteration in where?

Comment (squat) [reply projects/fal/src/fal/app.py:255]: It's really a redesign to use that since it's more idiomatic

Comment (isidentical) [reply projects/fal/src/fal/app.py:255]: ah, you mean the queue not EOF'ing with none but instead we raising a StopIteration inside the loop. I dont really touch that part of the code but i get the idea 👍🏻

Comment (chamini2) [inline projects/fal/src/fal/app.py:255]: can this `emitter.cancel` be called before all `outgoing_messages` are processed, leaving messages in the queue?

Comment (isidentical) [reply projects/fal/src/fal/app.py]: good catch!

**Review by turbo1912 — APPROVED** [review]

**Pull request merged**